src/software/ai/hl/stp/tactic/crease_defender/crease_defender_fsm.cpp

```diff
@@ -91,18 +106,39 @@ void CreaseDefenderFSM::blockThreat(
     TbotsProto::BallCollisionType ball_collision_type =
         TbotsProto::BallCollisionType::ALLOW;
     if ((event.common.world_ptr->ball().position() - destination).length() <
-        (event.common.robot.position() - destination).length())
+        (robot_position - destination).length())
     {
         ball_collision_type = TbotsProto::BallCollisionType::AVOID;
     }
 
+    AutoChipOrKick auto_chip_or_kick{AutoChipOrKickMode::OFF, 0};
+    auto goal_line_segment =
+        Segment(event.common.world_ptr->field().friendlyGoal().posXPosYCorner(),
+                event.common.world_ptr->field().friendlyGoal().posXNegYCorner());
+    Ray robot_shoot_ray = Ray(robot_position, robot_orientation);
+    std::vector<Point> goal_intersections =
+        intersection(robot_shoot_ray, goal_line_segment);
+    Stadium threat_zone = Stadium(
+        robot_position, Vector::createFromAngle(robot_orientation).normalize(1), 0.1);
+    double robot_to_net_m =
+        distance(robot_position, event.common.world_ptr->field().friendlyGoal().centre());
+
+    if (goal_intersections.empty() &&
+        CreaseDefenderFSM::isAnyEnemyInZone(event, threat_zone) &&
+        robot_to_net_m <= event.common.world_ptr->field().totalYLength() / 2)
+    {
+        // Autochip only if the robot is not facing the net, there is an enemy in front,
+        // and robot is close to net
```

Comment:
This comment might be better if it's placed a bit higher up. I was trying to understand all of the logic you had above, and just at the very end did I see the comment.